src/logger.js

```diff
@@ -31,6 +31,8 @@ const logger = {
     logMessage("ERROR", message, detail);
   },
   logFile(filename) {
+    loggedMessages.length = [];
```

Comment:
Clear the (shared!) global state before the bot executes. This fixes the bug where log messages from separate invocations were both ending up in the same file.

I've just spotted that the JavaScript code here is a bit mangled - it should be `arr.length = 0` or `arr = []` - I seem to have merged both approaches. However, JavaScript helpfully type converts `[]` to `0` so this actually works!

I'll fix this next time I have some more features / bugs